macro/src/lib.rs

```diff
@@ -21,3 +22,28 @@ pub fn write_html(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 pub fn derive_html_component(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
 	component::derive_html_component(input.into()).into()
 }
+
+/// makes a function that returns std::fmt::Result into a component function, argument to the
+/// attribute is the name of the struct that will be generated
+/// usage:
+/// ```rust
+/// #[component(MyComponent)]
+/// pub fn my_component(title: String) -> std::fmt::Result {
+///		component_html!(
+///			<div>{title}</div>
+///		)
+/// }
+/// ```
+#[proc_macro_attribute]
+pub fn component(
+	attr: proc_macro::TokenStream,
+	input: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+	utils::component(attr.into(), input.into()).into()
+}
+
+/// use inside component function, expands to `write_html!(formatter, #macro_input)`
```

Comment:
```suggestion
/// Use inside component functions, expands to `write_html!(formatter, #macro_input)`
```

I'm not convinced this macro is needed. I'll write some more about that in the general review comment - but if it's going to be kept, at least the documentation-comment should be formatted properly.